torch/distributed/distributed_c10d.py

```diff
@@ -393,7 +393,13 @@ def init_process_group(backend,
             the process group. Default value equals 30 minutes.
             This is applicable for the ``gloo`` backend. For ``nccl``, this is
             applicable only if the environment variable ``NCCL_BLOCKING_WAIT``
-            is set to 1.
+            or ``NCCL_ASYNC_ERROR_HANDLING`` is set to 1. When
+            ``NCCL_BLOCKING_WAIT`` is set, this is the duration for which the
+            process will block and wait for collectives to complete before
+            throwing an exception. When ``NCCL_ASYNC_ERROR_HANDLING`` is set,
+            this is the duration after which collectives will be aborted
+            asynchronously and the process will crash. Only one of these two
```

Comment:
Enhance this comment to mention when to use which argument. Ex: `NCCL_BLOCKING_WAIT` will provide errors to the user which can be caught and handled but due to the blocking nature it has a perf overhead. `NCCL_ASYNC_ERROR_HANDLING` doesn't have any perf overhead, but crashes the process on errors without giving the user any chance to handle errors. This is done since CUDA execution is async and it is no longer safe to continue executing user code since failed async NCCL operations might result in subsequent CUDA operations to run on corrupted data.